content/docs/patterns/no-series/index.md

```diff
@@ -0,0 +1,186 @@
++++
+title = "No. Series"
+tags = ["AL"]
+categories = ["Pattern"]
++++
+
+_Created by Microsoft, Described by Jeremy Vyska (Spare Brained Ideas)_
+
+## Abstract
+
+The "Number Series" system is used extensively to provide numbers to master records, documents, and other transactions through Microsoft Dynamics 365 Business Central.
+
+## Description
+
+At the heart of things, the Number Series engine allows users to define structure for a sequential numeric or alphanumeric string (collectively referred to as a 'number series'), then assign that structure to different parts of the system.
+
+Typically, one creates a single number series for each _type_ of data entity.  For example, Customers or Sales Orders each could have a series defined so that all new Customers or Sales Orders get a new number automatically.
+
+The Number Series system serves a few ancillary roles:
+
+- maintains the usage information to know when the last number was generated and on which date 
+- allows for date driven structures, so that different periods may have different structures
+- allows control of if manual entries are or are not permitted
+- allow for incrementing in different steps (+1 each time or +1000 each time)
+- warn users as a series is running out of numbers
+- control if any gaps in a series are permitted (as some regional laws do not allow skipping)
+
+This is many roles, features, and controls for generation of a single field so the implementation of this can seem difficult at first.
+
+{{% alert title="Note" color="info" %}}
+One additional (and somewhat optional) feature in the Number Series engine allows multiple sequences per type, called **Relationships**.  For example, different numbers for Items that are finished goods versus raw materials.  This requires additional hooks on the Page.
+{{% /alert %}}
+
+## Usage in Business Central
+
+To understand an example use in the Base App, the Customer data entity is a good choice.  
+
+Implementation to connect the Customer "No." field to the Number Series engine is done at the table level.  The Customer table contains:
+
+A field to contain the number (typically the primary key), which will be Code type, length of 20:
+
+```AL
+field(1; "No."; Code[20])
+{
+    Caption = 'No.';
+
+    trigger OnValidate()
+    begin
+        [...]
```

Comment:
Should we add the code that validates that a manual number is OK?

Reply:
Added